### DISCUSSION TITLE
express-rate-limit store support and maintenance

### DISCUSSION THREAD

**Discussion by nfriedly** [nfriedly/express-rate-limit#374] [General]
Hi @wyattjoh, @linyows, @2do2go, and @JGLTechnologies!

Firs of all, thank you for creating [rate-limit-redis](https://github.com/wyattjoh/rate-limit-redis), [rate-limit-memcached](https://github.com/linyows/rate-limit-memcached), [rate-limit-mongo](https://github.com/2do2go/rate-limit-mongo), and [precise-memory-rate-limit](https://github.com/JGLTechnologies/precise-memory-rate-limit) respectively.

I had an idea that I wanted to propose: would you like to transfer your stores to the express-rate-limit org and have @gamemaker1 and I share in the support and maintenance of them?

It occurred to me that it might feel like I was trying to take something away from you, which is not the goal. So if you'd prefer, we'd also be happy to help by contributing to the repos in their current locations, via PRs, code reviews, etc.

(For context, we've been slowly making some improvements to express-rate-limit to address longstanding issues, warn about common misconfigurations, and generally help users "fall into the pit of success", and I think some of these could be extended to the stores also.)

This is an individual decision for each of you, but I thought it might be good to start a discussion here with everyone and see where it takes us.

All the best!
-Nathan

**Comment by linyows** [comment]
`rate-limit-memcached` has not been maintained for a long time, so I agree with your proposal.

**Reply by nfriedly** [reply]
Ok, to transfer the github project, you'll 

1. Go to https://github.com/linyows/rate-limit-memcached/settings
2. Scroll down to the Danger Zone
3. Click the red Transfer button
4. Choose 'Specify an organization name'
5. Type in `express-rate-limit`
6. In the confirmation box type in `linyows/rate-limit-memcached`
7. Click the red 'I understand, transfer this repository' button

And, then, so that we can update the npm release, on a command line

1. Check if your logged into npm with the command `npm whoami`
    * It should say `linyows`, if not you'll want to log in with `npm login`
2. Run the following commands to grant Vedant and myself permission to publish updates
   ```sh
   npm owner add gamemaker1 rate-limit-memcached
   npm owner add nfriedly rate-limit-memcached
   ```
   * You may be prompted to authenticate again after each command.

**Reply by linyows** [reply]
@nfriedly I get this message: `You don’t have the permission to create public repositories on express-rate-limit` Please grant me for create repository on express-rate-limit. Or invite me to organization.

**Reply by gamemaker1** [reply]
@linyows I invited you to the organisation, you should be able to transfer the repo now

**Reply by linyows** [reply]
Thanks! I transferred a repository successfully.

**Reply by linyows** [reply]
I invited you to the npm package.

**Reply by gamemaker1** [reply]
Thank you!!

**Reply by nfriedly** [reply]
Thank you both!  Sorry I missed the step of inviting you to the org @linyows.

**Comment by wyattjoh** [comment]
I agree that I haven't had a lot of time with my current role to keep [rate-limit-redis](https://github.com/wyattjoh/rate-limit-redis) maintained. I'm happy to have it join y'alls organization and have CI be enabled for publishing to the npm repository from it.

**Reply by nfriedly** [reply]
That's fair, I just invited you to the express-rate-limit org

**Reply by wyattjoh** [reply]
Transferred and invited!

**Reply by gamemaker1** [reply]
Thank you!!
